src/apphosting/index.ts

```diff
@@ -65,15 +65,21 @@ export async function doSetup(
 
   location =
     location ||
-    ((await promptOnce({
+    (await promptOnce({
       name: "region",
       type: "list",
       default: DEFAULT_REGION,
       message:
         "Please select a region " +
         `(${clc.yellow("info")}: Your region determines where your backend is located):\n`,
       choices: allowedLocations.map((loc) => ({ value: loc })),
-    })) as string);
+    }));
+
+  if (!location) {
+    throw new FirebaseError(
+      `Location must be provided. Valid choices are ${allowedLocations.join(", ")}`,
+    );
+  }
```

Comment:
Yeah this was mainly to appease the typescript linter which was insisting `location` could be `null` because the return value of `promptOnce` is not type safe. Alternatively we could just do a type assertion here.